runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4002,7 +4002,7 @@ J9::Z::TreeEvaluator::generateTestAndReportFieldWatchInstructions(TR::CodeGenera
       if (isResolved)
          {
          fieldClassReg = cg->allocateRegister();
-         if (!(cg->needClassAndMethodPointerRelocations()))
+         if (!(cg->needClassAndMethodPointerRelocations()) && cg->canUseRelativeLongInstructions(reinterpret_cast<int64_t>((static_cast<TR::J9WatchedStaticFieldSnippet *>(dataSnippet)->getFieldClass()))))
```

Comment:
Why are we hitting this path for JITServer at all? Shouldn't `needClassAndMethodPointerRelocations` return `true` for JITServer?

Reply:
It returns `false` for regular remote compilations (i.e. when we have a `TR_J9ServerVM`). We usually have that information from the client. We just can't use the relative path here. I figured taking the `AOT` path here for JITServer also made sense since this likely won't have much impact on field watch performance. I think these queries sufficiently describe why we can't take the forward path of this branch for regular remote compilations.

Reply:
Ok, thanks for explaining. Could we make a small cleanup change while in the area?

```
         fieldClassReg = cg->allocateRegister();
         J9Class *fieldClass = static_cast<TR::J9WatchedStaticFieldSnippet *>(dataSnippet)->getFieldClass();
         if (!cg->needClassAndMethodPointerRelocations() && cg->canUseRelativeLongInstructions(reinterpret_cast<int64_t>(fieldClass)))
            {
            // For non-AOT (JIT and JITServer) compiles ...
            TR_ASSERT_FATAL(...
            }
```

Reply:
Done here: 27f02c3